lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -55,15 +52,22 @@ def call(env)
           request_span.set_error(e)
           raise e
         ensure
+          # the source of truth in Rack is the PATH_INFO key that holds the
+          # URL for the current request; some framework may override that
+          # value, especially during exception handling and because of that
+          # we prefer using the `REQUEST_URI` if this is available.
```

Comment:
actually `PATH_INFO` must include the request URL as defined in Rack docs: http://www.rubydoc.info/github/rack/rack/file/SPEC#The_Environment

We may double check on Rails why it's happening. This is just a reasonable default because we know about this exception, but definitely something to take in consideration if it happens also for other frameworks.